core/domain/user_query_domain_test.py

```diff
@@ -29,11 +29,11 @@
 class UserQueryTests(test_utils.GenericTestBase):
     """Test for the UserQuery."""
 
-    def setUp(self):
+    def setUp(self) -> None:
         super(UserQueryTests, self).setUp()
         self.signup(self.CURRICULUM_ADMIN_EMAIL, self.CURRICULUM_ADMIN_USERNAME)
-        self.user_id = self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL)
-        self.user_query_params = user_query_domain.UserQueryParams(
+        self.user_id = self.get_user_id_from_email(self.CURRICULUM_ADMIN_EMAIL) # type: ignore[no-untyped-call]
+        self.user_query_params = user_query_domain.UserQueryParams( # type: ignore[call-arg]
```

Comment:
Why this type ignore?

Reply:
So all the statements i marked with `# type: ignore[no-untyped-call]` is because these functions are yet to be annotated . Thus these functions return the error `untyped function` and needed to be skipped for now .

And for all other ignore statements , it is because it is a test file and we are deliberately passing wrong values to test that function . After going through the oppia wiki , it instructed to ignore these statements with appropriate statements .

---

![2022 02 14-19 50 37 screenshot](https://user-images.githubusercontent.com/63491234/153881447-a6ad8cd6-717e-4187-afd5-4aa1e320e3b3.png)

Reply:
I think you can remove some of the `isinstance` checks from the `validate` function, as with typing they are no longer that much needed.

Reply:
Why is this ignore still needed?

Reply:
for the one with [no-untyped-call] it is because the functions defined in UserQueryTests in test_utils.py are still not annotated . Thus we need this ignore for skipping that ensure typed check .

for the other one [call-arg] it is because the fields/attributes ( attribute_names ) for namedtuple UserQueryParams are generated dynamically . Thus mypy also cannot predict the attributes / args / fields for UseryQueryParams before runtime . Thus we also need to ignore that mypy check there .